NeoML/src/TraditionalML/SparseFloatMatrix.cpp

```diff
@@ -56,28 +56,51 @@ CSparseFloatMatrix::CSparseFloatMatrixBody::CSparseFloatMatrixBody( int height,
 
 CSparseFloatMatrix::CSparseFloatMatrixBody::CSparseFloatMatrixBody( const CSparseFloatMatrixDesc& desc ) :
 	RowsBufferSize( desc.Height ),
-	ElementsBufferSize( desc.Height == 0 ? 0 : desc.PointerE[desc.Height - 1] ),
-	ElementCount( desc.Height == 0 ? 0 : desc.PointerE[desc.Height - 1] )
+	ElementsBufferSize( desc.Height == 0 ? 0 : ( desc.Columns != nullptr ? desc.PointerE[desc.Height - 1] : 0 ) ),
+	ElementCount( desc.Height == 0 ? 0 : ( desc.Columns != nullptr ? desc.PointerE[desc.Height - 1] : 0 ) )
 {
 	NeoAssert( RowsBufferSize >= 0 );
 	NeoAssert( ElementsBufferSize >= 0 );
 	Desc.Height = desc.Height;
 	Desc.Width = desc.Width;
 
-	ColumnsBuf.SetSize( ElementsBufferSize );
-	ValuesBuf.SetSize( ElementsBufferSize );
 	BeginPointersBuf.SetSize( RowsBufferSize );
 	EndPointersBuf.SetSize( RowsBufferSize );
+	Desc.PointerB = BeginPointersBuf.GetPtr();
+	Desc.PointerE = EndPointersBuf.GetPtr();
+	if( desc.Columns == nullptr ) {
+		for( int i = 0; i < desc.Height; ++i ) {
```

Comment:
for( int i = 0; i < d.H * d.W; i++ ) {
  if( desc.Value[i] != 0 ) {
    ++ElementBufferSize;
  }
}

Reply:
indeed... fixed